octoprint_mrbeam/materials.py

```diff
@@ -152,17 +57,46 @@ def _save(self, force=False):
                     indent="  ",
                     allow_unicode=True,
                 )
-            self.custom_materials_loaded = True
-            self._logger.debug(
-                "Saved %s custom materials (in total) to file %s",
-                len(self.custom_materials),
-                self.custom_materials_file,
-            )
-        except:
-            self._logger.exception(
-                "Exception while writing custom materials to file %s",
-                self.custom_materials_file,
+            self._logger.info("{} custom materials saved".format(len(self.custom_materials)))
+        except Exception as e:
+            self._logger.exception("Exception while saving custom materials: {}".format(e))
+
+    def get_custom_materials_for_laser_cutter_mode(self):
```

Comment:
Since the method doesn't have any parameters and just gets the custom material (with the current mode as it does with the current laser head) then I would not have it name "for_laser_cutter_mode"

Reply:
The laser head definition is still in the frontend but we will port it as some point to the backend and it should have only 1 method to get custom materials regardless of the underlaying current conditions

Reply:
maybe something like get_custom_materials_by_settings, more general